docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,129 @@
+# ChiselEnum
+
+The ChiselEnum type is an easy way to get away from value encodings and reduce the chance of error when encoding muxes, opcodes, and functional unit operations. As opposed to regular Enums, ChiselEnums allow for function names to be explicitly passed through Bundles like IO. 
+
+## Importing the ChiselEnum module
+
+```scala mdoc:silent
+import chisel3._
+import chisel3.experimental.ChiselEnum
+```
+
+## Functionality and Examples
+
+Below we see ChiselEnum being used as mux select for a RISC-V core. While wrapping the object in a package is not required, it is highly recommended as it allows for the type to be used in multiple files more easily. 
+
+```scala mdoc:silent
+package CPUTypes {
+    object AluMux1Sel extends ChiselEnum {
+        val rs1out, pcout = Value
+        /**
+            "rs1out" -> 0.U,
+            "pcout"  -> 1.U
+        */
+    }
+}
+```
+
+Here we see a mux using the AluMux1Sel to select between different inputs. 
+
+```scala mdoc:silent
+import CPUTypes.AluMux1Sel
+import CPUTypes.AluMux1Sel._
+
+class AluMux1File(val dl_size: Int) extends Module {
+    val io = IO(new Bundle {
+        val aluMux1Sel =  Input( AluMux1Sel() )
+        val rs1Out     =  Input(Bits(dl_size.W))
+        val pcOut      =  Input(Bits(dl_size.W))
+        val aluMux1Out = Output(Bits(dl_size.W))
+    })
+
+    // Default value for aluMux1Out
+    io.aluMux1Out := 0.U
```

Comment:
shouldn't the default be one of the `ChiselEnum` types?

Reply:
It can be a default value depends on whether logic is being simplified or error checked. Something like `DontCare` or `0.U` can be used for checking that unknown selects are used. I believe that this is a matter of implementation.

Reply:
Oh, I misread the example, the type is on the select not the data.

Reply:
Correct!